core/templates/dev/head/collection_editor/CollectionEditor.js

```diff
@@ -0,0 +1,213 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Primary controller for the collection editor page.
+ *
+ * @author mgowano@google.com (Abraham Mgowano)
+ */
+
+// TODO(bhenning): These constants should be provided by the backend.
+oppia.constant(
+  'COLLECTION_DATA_URL', '/collection_handler/data/<collection_id>');
```

Comment:
Optional nit: COLLECTION_DATA_URL_TEMPLATE or COLLECTION_DATA_UNINTERPOLATED_URL? Ditto below. (As it is, it's not really a URL.)